packages/react-components/react-color-picker-preview/library/docs/Spec.md

```diff
@@ -2,62 +2,458 @@
+| shape      | `square`, `rounded` | `square`  | Sets shape                             |
+| step       | `number`            | 1         | Step for arrow navigation              |
+| customStep | `number`            | 3         | Fast navigation                        |
+
+Input fields with color values will be in `renderUtils`. It will contain default ColorPicker which has all colors represented and a preview swatch. This functionality is similar to V8.
+
+```ts
+import { ColorPickerLayout } from '@fluentui/react-color-picker-preview';
+```
+
+```tsx
+export const Default = () => {
+  const COLOR = 'rgba(0, 255, 170, 1)';
+  const [selectedColor, setSelectedColor] = React.useState(COLOR);
+  const handleChange: ColorPickerOnSelectEventHandler = (_, data) => {
+    setSelectedColor(data.selectedColorHex);
+  };
+
+  return (
+    <>
+      <ColorPickerLayout
+        color={selectedColor}
+        onChange={handleChange}
+        showAlphaSlider={true}
+        showPreview={true}
+        channels={['hex', 'rgb']}
+      />
+      <div style={{ backgroundColor: `${selectedColor}` }} />
+    </>
+  );
+};
+```
+
+![ColorPickerLayout](./assets/color-picker-layout.jpg)
+
+Validation of color fields will be in `utils`. Validation should not allow entering incorrect values to the input fields.
+
+For color manipulation and conversion `tinycolor` library will be used.
+
+| Slots | Values | Default | Description                 |
+| ----- | ------ | ------- | --------------------------- |
+| root  | `div`  | `div`   | The root of the ColorPicker |
+
+### ColorArea
+
+| Property   | Values              | Default   | Purpose                                |
+| ---------- | ------------------- | --------- | -------------------------------------- |
+| customStep | `number`            | 3         | Fast step for the slider               |
+| onChange   | `function`          | undefined | Callback called when color is selected |
+| shape      | `square`, `rounded` | `square`  | Sets shape                             |
+| step       | `number`            | 1         | Step for the slider                    |
+| valueX     | `string`            |           | Value of the slider on X axis          |
+| valueY     | `string`            |           | Value of the slider on Y axis          |
+
+| Slots   | Values  | Default | Description                                                         |
+| ------- | ------- | ------- | ------------------------------------------------------------------- |
+| root    | `div`   | `div`   | The root of the ColorArea element                                   |
+| thumb   | `div`   | `div`   | The draggable icon used to select a given value from the ColorArea. |
+| sliderX | `input` | `input` | Input for X axis                                                    |
+| sliderY | `input` | `input` | Input for Y axis                                                    |
+
+### ColorSlider
+
```

Comment:
Probably, it would be good to support custom gradients. Then `channel` prop will be needed.